securitas/tests/unit/controller/test_password.py

```diff
@@ -9,3 +10,105 @@ def test_password_reset(client):
     page = BeautifulSoup(result.data, 'html.parser')
     assert page.title
     assert page.title.string == 'Password Reset - The Fedora Project'
+
+
+def test_non_matching_passwords(client):
+    """Verify that passwords that dont match are caught"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "jbloggs",
+            "current_password": "sdsds",
+            "password": "password1",
+            "password_confirm": "password2",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    password_input = page.select("input[name='password']")[0]
+    assert 'invalid' in password_input['class']
+    helper_text = password_input.find_next("span", class_="helper-text")
+    assert helper_text["data-error"] == "Passwords must match"
+
+
+@pytest.mark.vcr()
+def test_password(client):
+    """Verify that current password must be correct"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "admin",
+            "current_password": "1",
+            "password": "LongSuperSafePassword",
+            "password_confirm": "LongSuperSafePassword",
+        },
+        follow_redirects=True,
+    )
+    assert result.status_code == 200
+    page = BeautifulSoup(result.data, 'html.parser')
+    password_input = page.select("input[name='current_password']")[0]
+    assert 'invalid' in password_input['class']
+    helper_text = password_input.find_next("span", class_="helper-text")
+    assert helper_text["data-error"] == "The old password or username is not correct"
+
+
+@pytest.mark.vcr()
+def test_time_sensitive_password_policy(client, dummy_user):
+    """Verify that new password policies are upheld"""
+    result = client.post(
+        '/password-reset',
+        data={
+            "username": "dummy",
+            "current_password": "dummy_password",
+            "password": "somesupersecretpassword",
+            "password_confirm": "somesupersecretpassword",
+        },
+        follow_redirects=True,
+    )
+    page = BeautifulSoup(result.data, 'html.parser')
+    password_input = page.select("input[name='password']")[0]
+    assert 'invalid' in password_input['class']
+    helper_text = password_input.find_next("span", class_="helper-text")
+    assert helper_text["data-error"] == "Constraint violation: Too soon to change password"
```

Comment:
That's going to depend on the order in which the tests are run: some other test must have changed the password before. Not sure how to work around that, though.

Reply:
I think this works because in the fixtures file you have `ipa.change_password()` right after the dummy user is created. It works well because its also the reason I needed to disable the time policy further down. The only other thing I can think of is disable the time policy, change the password again, re-enable the time policy and then run this test.

Reply:
Ah good point, the password is always changed after user creation. Could you add a comment to explain that please?